lib/src/native/realm_core.dart

```diff
@@ -1956,7 +1926,7 @@ extension on realm_sync_error {
 
 extension on Pointer<realm_sync_error_code_t> {
   SyncError toSyncError() {
-    final message = ref.message.cast<Utf8>().toRealmDartString(freeRealmMemory: true)!;
+    final message = ref.message.cast<Utf8>().toDartString();
```

Comment:
are we sure we don't need to free the realm memory here

Reply:
Same as above.